File: Flow/Flow/app/src/main/java/com/example/flow/Home_Main.java
package com.example.flow;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.viewpager.widget.ViewPager;


import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import com.google.android.material.navigation.NavigationView;
import com.google.android.material.tabs.TabItem;
import com.google.android.material.tabs.TabLayout;

public class Home_Main extends AppCompatActivity {

    /// this activity function as the first activity the user will see
    // there will be three tabs that will be connected to three fragments
    //


    //tab instances
    TabLayout tb_layout;
    TabItem TabOne,TabTwo,TabThree;
    App_Accessor SectionsAccessor;

    //navigation and side draw
    ViewPager tabs_view;
    DrawerLayout drawerLayout;
    ActionBarDrawerToggle Toggle;
    NavigationView navigationView;
    ImageView imageView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home__main);
        settingVariables();


        tb_layout = findViewById(R.id.tabs_Layout);
        SectionsAccessor = new App_Accessor( getSupportFragmentManager(),tb_layout.getTabCount(),tb_layout.getTabCount() );

        tabs_view = findViewById(R.id.viewPager);
        //uncomment shouild we add the tool bar back
        //Toolbar ToolBar = findViewById(R.id.toolbar);
       // setSupportActionBar(ToolBar);

        tabs_view.setAdapter(SectionsAccessor);
        tb_layout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                tabs_view.setCurrentItem(tab.getPosition());
            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
            }
            @Override
            public void onTabReselected(TabLayout.Tab tab) {
            }
        });
        tabs_view.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tb_layout));






    }
    //methods will go here /////////////////////

    private void settingVariables()
    {
        //void function to set variable to be used through out activity such as findbyid


    }

}

File: Studio_App2/Studio_App/app/src/main/java/com/android/example/studio_app/Home_Main.kt
package com.android.example.studio_app

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.viewpager.widget.ViewPager
import com.android.example.studio_app.pageAdapter.HomePageFragmentAdapter
import com.google.android.material.tabs.TabLayout
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase

class Home_Main : AppCompatActivity() {
    // viewPager




    private companion object {
        private lateinit var tabViewAdapter: HomePageFragmentAdapter
        private lateinit var viewPager: ViewPager
        private lateinit var tabLayout: TabLayout
        private const val TAG = "Home_main"
    }



    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home_main)
        auth = Firebase.auth


        setUpViewPager()

    }






    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        //Inflates the menu
        menuInflater.inflate(R.menu.logout_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if(item.itemId == R.id.miLogout){

            Log.i(TAG, " Logout")
            auth.signOut()
            val logOutIntent = Intent(this, MainActivity::class.java)
            logOutIntent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(logOutIntent)

        }
        return super.onOptionsItemSelected(item)
    }







    private fun setUpViewPager() {
        setUpTabLayout()

    }

    private fun setUpTabLayout() {
        // Tab Layout
        tabLayout = findViewById(R.id.tabs_Layout)

        // Create an instance of the HomePageFragmentAdapter
        tabViewAdapter = HomePageFragmentAdapter(this, supportFragmentManager, tabLayout.tabCount)
        // make an instance of the viewPager
        viewPager = findViewById(R.id.viewPager)

        viewPager.adapter = tabViewAdapter

        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))




        tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab?) {
                viewPager.currentItem = tab!!.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab?) {

            }

            override fun onTabReselected(tab: TabLayout.Tab?) {

            }

        })

    }
}